src/Core/Entity/Image/InputImage.php

```diff
@@ -38,6 +42,33 @@ public function __construct(OptionsBag $optionsBag, string $sourceImageUrl)
 
         $this->sourceImagePath = $optionsBag->hashOriginalImageUrl($this->sourceImageUrl);
         $this->saveToTemporaryFile();
+
+        // Store the source file mime type.
+        $this->sourceFileMimeType = finfo_file(
+            finfo_open(FILEINFO_MIME_TYPE),
+            $this->sourceImagePath
+        );
+
+        // If the source file has a video mime type, generate a full resolution
```

Comment:
Could you please move the logic behind video processing to a separate Processor Class? 
you can check what faceDetection class how it was implemented and called from ImageHandler class.
It'll make the code more readable and simple.
Doing this you'll fix also the codeclimate issues.

Reply:
It's been moved to VideoProcessor.php, still have a couple of code climate issues, not sure where to move those.